File: Classwork/Classwork09/FactorialRecursor.java
/**
 * filename: FactorialRecursor.java
 * purpose: demonstrate recursion using factorial
 * @author: Dr. Johnson
 * @date: 2020-10-25
 */

public class FactorialRecursor {
   public int factorial( int n ) {
      System.out.println( "N is: " + n );
      if( n == 0 ) {
         return 1;
      } else {
         return (n * factorial( n - 1 ));
      }
   }

   public static void main( String[] args ) {
      FactorialRecursor fr = new FactorialRecursor();
      int count = Integer.parseInt( args[0] );
      System.out.println( "The value of " + count + "! is: " + fr.factorial( count ) );
      System.exit( 0 );
   }
}

File: Classwork/classwork09/FactorialRecursor.py
### 
# filename: FactorialRecursor.py 
# purpose:  demonstrate recursion
# author:   Dr. Johnson 
# date:     2023-02-20
### 

import sys

def factorial( n ):
#   print( "   N is: ", n )
   if( n <= 0 ):
      return 1
   else:
      return( n * factorial( n - 1 ) )

def main():
   if len(sys.argv) == 1:
      count = int( input( "\n   Enter an integer number: " ) )
   else:
      count = int( sys.argv[1] )
   print( "\n   The value of ", count, " factorial is: ", factorial( count ) )

main()
